stubs/tensorflow/tensorflow/train/__init__.pyi

```diff
@@ -0,0 +1,77 @@
+from _typeshed import Incomplete
+from collections.abc import Callable
+from typing import Any, TypeVar
+from typing_extensions import Self
+
+import numpy as np
+import tensorflow as tf
+from tensorflow.core.example.example_pb2 import Example as Example
+from tensorflow.core.example.feature_pb2 import (
+    BytesList as BytesList,
+    Feature as Feature,
+    Features as Features,
+    FloatList as FloatList,
+    Int64List as Int64List,
+)
+from tensorflow.core.protobuf.cluster_pb2 import ClusterDef as ClusterDef
+from tensorflow.core.protobuf.tensorflow_server_pb2 import ServerDef as ServerDef
+from tensorflow.python.trackable.base import Trackable
+
+class CheckpointOptions:
+    experimental_io_device: None | str
+    experimental_enable_async_checkpoint: bool
+    # experimental_write_callbacks: None | list[Callable[[str], Any] | Callable[[], Any]]
```

Comment:
Are these also due to be added in a future release? I'm OK with leaving them commented out for now, but please put a comment somewhere in the stub about it, so we don't forget why there is this commented out code here.

Reply:
Yes, looking at the [2.12 doc](https://www.tensorflow.org/versions/r2.12/api_docs/python/tf/train/CheckpointOptions) they are not present.
One of them is already in 2.15, two are not present in a release yet but are present in [the tensorflow code](https://github.com/tensorflow/tensorflow/blame/8f1e353db44fb5d9b2ddfd762b475aefa23ac49f/tensorflow/python/checkpoint/checkpoint_options.py#L49).

I've removed the ones not present in 2.15, and added comments for the rest in 2ad4be9a2.